contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,933 @@
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        recipient: address(this),
+                        deadline: block.timestamp,
+                        sqrtPriceX96: 0
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+            tokenId = _mintedTokenId;
+        } else {
+            (
+                ,
+                uint256 __wethAmountSupplied,
+                uint256 __oethbAmountSupplied
+            ) = positionManager.increaseLiquidity(
+                    /** amount0Min & amount1Min are left at 0 because slippage protection is ensured by the
+                     * _checkForExpectedPoolPrice
+                     */
+                    INonfungiblePositionManager.IncreaseLiquidityParams({
+                        tokenId: tokenId,
+                        amount0Desired: _wethBalance,
+                        amount1Desired: _oethbRequired,
+                        amount0Min: 0,
+                        amount1Min: 0,
+                        deadline: block.timestamp
+                    })
+                );
+
+            _wethAmountSupplied = __wethAmountSupplied;
+            _oethbAmountSupplied = __oethbAmountSupplied;
+        }
+
+        _updateUnderlyingAssets();
+        emit LiquidityAdded(
+            _wethBalance, // wethAmountDesired
+            _oethbRequired, // oethbAmountDesired
+            _wethAmountSupplied, // wethAmountSupplied
+            _oethbAmountSupplied, // oethbAmountSupplied
+            tokenId, // tokenId
+            underlyingAssets
+        );
+
+        // burn remaining OETHb
+        _burnOethbOnTheContract();
+        positionManager.approve(address(clGauge), tokenId);
+        clGauge.deposit(tokenId);
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /**
+     * @dev Check that the Aerodrome pool price is within the expected
+     *      parameters.
+     *      This function ignores whether the strategy contract has liquidity
+     *      position in the pool.
+     */
+    function _checkForExpectedPoolPrice() internal {
+        uint160 _currentPrice = getPoolX96Price();
+        // check we are in inspected tick range
+
+        if (_currentPrice < sqrtRatioX96Tick0 ||
+            _currentPrice > sqrtRatioX96Tick1) {
+
+            int24 currentTick = getCurrentTradingTick();
+            revert OutsideExpectedTickRange(
+                currentTick,
+                lowerTick,
+                upperTick
+            );
+        }
+
+        // If token addresses were reversed estimateAmount0 would
+        // be required here
+        uint256 _normalizedWethAmount = 1 ether;
+        uint256 _correspondingOETHAmount = helper.estimateAmount1(
+            _normalizedWethAmount,
+            address(clPool),
+            _currentPrice,
+            lowerTick,
+            upperTick
+        );
+
```

Comment:
⚪️ I still wonder how much simpler this method would be if we just precalced a low price ratio bound and a high price ratio bound in a single set method, and just checked that the price was inside those bounds. This instead of storing poolWethShare and poolWethShareVarianceAllowed.

Reply:
posting relevant discussion for reference: https://github.com/OriginProtocol/origin-dollar/pull/2184#discussion_r1729172752